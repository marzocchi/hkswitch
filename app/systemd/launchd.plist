<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>

<key>Label</key>
<string>io.mrz.hkswitch.{{.Description}}</string>

<key>RunAtLoad</key>
<true/>

{{ if .Env }}<key>EnvironmentVariables</key>
<dict>{{ range .Env }}
<key>{{.Name}}</key>
<string><![CDATA[{{.Value}}]]></string>{{ end }}
</dict>{{ end }}

<key>WorkingDirectory</key>
<string>{{ .WorkingDir }}</string>

<key>ProgramArguments</key>
<array>
{{ range .Command }}<string>{{ . }}</string>{{ end }}
</array>

</dict>
</plist>
