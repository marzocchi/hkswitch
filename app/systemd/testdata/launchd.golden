<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>

<key>Label</key>
<string>io.mrz.hkswitch.test</string>

<key>RunAtLoad</key>
<true/>

<key>EnvironmentVariables</key>
<dict>
<key>VARNAME1</key>
<string><![CDATA[VARVALUE1]]></string>
<key>VARNAME2</key>
<string><![CDATA[VARVALUE2]]></string>
</dict>

<key>WorkingDirectory</key>
<string>/var/run</string>

<key>ProgramArguments</key>
<array>
<string>cmd</string><string>arg</string>
</array>

</dict>
</plist>
